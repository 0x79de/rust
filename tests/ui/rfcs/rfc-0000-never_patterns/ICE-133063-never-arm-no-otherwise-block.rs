#![feature(never_patterns)]
#![feature(if_let_guard)]
#![allow(incomplete_features)]

fn split_last(_: &()) -> Option<(&i32, &i32)> {
    None
}

fn assign_twice() {
    loop {
        match () {
            (!| //~ ERROR: mismatched types
            !) if let _ = split_last(&()) => {} //~ ERROR a never pattern is always unreachable
            //~^ ERROR: mismatched types
            //~^^ WARNING: irrefutable `if let` guard pattern [irrefutable_let_patterns]
            _ => {}
        }
    }
}

fn main() {}
