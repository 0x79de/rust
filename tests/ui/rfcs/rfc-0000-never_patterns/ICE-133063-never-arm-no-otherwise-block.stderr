error: a never pattern is always unreachable
  --> $DIR/ICE-133063-never-arm-no-otherwise-block.rs:13:46
   |
LL |             !) if let _ = split_last(&()) => {}
   |                                              ^^
   |                                              |
   |                                              this will never be executed
   |                                              help: remove this expression

error: mismatched types
  --> $DIR/ICE-133063-never-arm-no-otherwise-block.rs:12:14
   |
LL |             (!|
   |              ^ a never pattern must be used on an uninhabited type
   |
   = note: the matched value is of type `()`

error: mismatched types
  --> $DIR/ICE-133063-never-arm-no-otherwise-block.rs:13:13
   |
LL |             !) if let _ = split_last(&()) => {}
   |             ^ a never pattern must be used on an uninhabited type
   |
   = note: the matched value is of type `()`

warning: irrefutable `if let` guard pattern
  --> $DIR/ICE-133063-never-arm-no-otherwise-block.rs:13:19
   |
LL |             !) if let _ = split_last(&()) => {}
   |                   ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this pattern will always match, so the guard is useless
   = help: consider removing the guard and adding a `let` inside the match arm
   = note: `#[warn(irrefutable_let_patterns)]` on by default

error: aborting due to 3 previous errors; 1 warning emitted

